src/plugins/timelion/server/deprecations.ts

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+import {
+  CoreStart,
+  SavedObjectsClient,
+  Logger,
+  RegisterDeprecationsConfig,
+  DeprecationsDetails,
+} from 'src/core/server';
+
+export const getTimelionSheetsCount = async (
+  savedObjectsClient: Pick<SavedObjectsClient, 'find'>
+) => {
+  const { total } = await savedObjectsClient.find({ type: 'timelion-sheet', perPage: 1 });
+  return total;
+};
+
+export const showWarningMessageIfTimelionSheetWasFound = async (
+  core: CoreStart,
+  logger: Logger
+) => {
+  const { savedObjects } = core;
+  const savedObjectsClient = savedObjects.createInternalRepository();
+  const count = await getTimelionSheetsCount(savedObjectsClient);
+  if (count > 0) {
+    logger.warn(
+      'Deprecated since 7.0, the Timelion app will be removed in 8.0. To continue using your Timelion worksheets, migrate them to a dashboard. See https://www.elastic.co/guide/en/kibana/current/create-panels-with-timelion.html.'
+    );
+  }
+};
+
+/**
+ * Deprecated since 7.0, the Timelion app will be removed in 8.0.
+ * To continue using your Timelion worksheets, migrate them to a dashboard.
+ *
+ *  @link https://www.elastic.co/guide/en/kibana/master/timelion.html#timelion-deprecation
+ **/
+export const getDeprecations: RegisterDeprecationsConfig['getDeprecations'] = async ({
+  savedObjectsClient,
+}) => {
+  const deprecations: DeprecationsDetails[] = [];
+  const count = await getTimelionSheetsCount(savedObjectsClient);
+
+  if (count > 0) {
+    deprecations.push({
+      message: `You have ${count} Timelion worksheets. The Timelion app will be removed in 8.0. To continue using your Timelion worksheets, migrate them to a dashboard.`,
+      documentationUrl:
+        'https://www.elastic.co/guide/en/kibana/master/dashboard.html#timelion-deprecation',
```

Comment:
The documentation link has changed. Can we use the correct one here? https://www.elastic.co/guide/en/kibana/current/create-panels-with-timelion.html

Reply:
Updated the link